tests/test_nhg.py

```diff
@@ -1379,6 +1617,507 @@ def test_nhgorch_label_route(self, dvs, testlog):
+            time.sleep(1)
+            assert(self.asic_db.get_entry(self.ASIC_NHG_STR, nhg_id)['SAI_NEXT_HOP_GROUP_ATTR_SELECTION_MAP'] == smap_id)
+            fvs = swsscommon.FieldValuePairs([('members', 'group2'), ('selection_map', 'cbfnhgmap4')])
+            self.cbf_nhg_ps.set('cbfgroup1', fvs)
+
+        # Test scenario:
+        # - create a NHG that points to a map that refers to more members than the group has
+        def create_cbf_invalid_nhg_map_test():
+            # Create an FC to NH index selection map that references 3 group members
+            fvs = swsscommon.FieldValuePairs([('0', '1'), ('1', '0'), ('2', '2')])
+            self.fc_to_nhg_ps.set('cbfnhgmap3', fvs)
+            self.asic_db.wait_for_n_keys(self.ASIC_NHG_MAP_STR, self.asic_nhg_maps_count + 2)
+
+            # Create a group that references this map. It doesn't get created.
+            fvs = swsscommon.FieldValuePairs([('members', 'group3,group2'),
+                                                ('selection_map', 'cbfnhgmap3')])
+            self.cbf_nhg_ps.set('cbfgroup3', fvs)
+            time.sleep(1)
+            assert(not self.nhg_exists('cbfgroup3'))
+
+        self.init_test(dvs, 4)
+
+        mainline_cbf_nhg_test()
+        data_validation_test()
+        update_cbf_nhg_members_test()
+        update_cbf_nhg_map_test()
+        delete_referenced_cbf_nhg_test()
+        create_route_inexistent_cbf_nhg_test()
+        update_deleting_cbf_nhg_test()
+        create_cbf_invalid_nhg_map_test()
+
+        # Delete the NHGs
+        self.cbf_nhg_ps._del('cbfgroup1')
+        self.nhg_ps._del('group2')
+        self.nhg_ps._del('group3')
+        self.asic_db.wait_for_n_keys(self.ASIC_NHG_STR, self.asic_nhgs_count)
+        self.asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, self.asic_nhgms_count)
+
+        update_cbf_nhg_inexistent_member_test()
+        update_cbf_nhg_inexistent_map_test()
+
+        # Coverage testing: Update the CBF NHG with a member that does not exist.
+        fvs = swsscommon.FieldValuePairs([('members', 'group3'), ('selection_map', 'cbfnhgmap2')])
+        self.cbf_nhg_ps.set('cbfgroup1', fvs)
+        time.sleep(1)
+
+        # Delete the NHGs
+        self.cbf_nhg_ps._del('cbfgroup1')
+        self.nhg_ps._del('group2')
+        self.asic_db.wait_for_n_keys(self.ASIC_NHG_STR, self.asic_nhgs_count)
+
+        # Delete the NHG maps
+        self.fc_to_nhg_ps._del('cbfnhgmap1')
+        self.fc_to_nhg_ps._del('cbfnhgmap2')
+        self.fc_to_nhg_ps._del('cbfnhgmap3')
+        self.fc_to_nhg_ps._del('cbfnhgmap4')
+        self.asic_db.wait_for_n_keys(self.ASIC_NHG_MAP_STR, self.asic_nhg_maps_count)
+
+        # Coverage testing: Delete inexistent CBF NHG
+        self.cbf_nhg_ps._del('cbfgroup1')
+
+    def test_nhg_map(self, dvs, testlog):
+
+        # Test scenario:
+        # - create a NHG map and assert the expected details
+        # - delete the map
+        def mainline_nhg_map_test():
```

Comment:
Test case for NHGMAp update - new NHG add/delete?

Reply:
Wasn't quite sure what you meant here, but I've added some bits to this test case

Reply:
@TACappleman  - could you please give pointer to the test case you mentioned?

Reply:
I've added an update of the map to mainline_nhg_map_test, and created and deleted a referencing NHG in that